wetlab/views.py

```diff
@@ -2441,230 +2410,119 @@ def record_samples(request):
+                    "error_message": core.core_config.ERROR_EMPTY,
                 },
             )
         else:
             # Check if all samples have same project and if project is in the DB
-            if core.utils.load_batch.project_validation(sample_batch_df, __package__):
+            validation_project, project_name = core.utils.load_batch.project_validation(
+                sample_batch_df, __package__
+            )
+            if validation_project["result"] == "NOK":
                 return render(
                     request,
                     "wetlab/record_sample.html",
                     {
                         "fields_info": fields_info,
-                        "error_message": core.utils.load_batch.project_validation(
-                            sample_batch_df, __package__
-                        ),
+                        "error_message": validation_result["error_message"],
                     },
                 )
-
-            sample_batch_df = core.utils.load_batch.heading_refactor(sample_batch_df)
-
             # Test if column names are valid
-            if core.utils.load_batch.validate_header(sample_batch_df):
+            sample_columns = list(sample_batch_df.columns.values)
+            validation_result = core.utils.load_batch.validate_header(
+                sample_columns, project_name
+            )
+            if validation_result["result"] == "NOK":
                 return render(
                     request,
                     "wetlab/record_sample.html",
                     {
                         "fields_info": fields_info,
-                        "error_message": core.utils.load_batch.validate_header(
-                            sample_batch_df
-                        ),
+                        "error_message": validation_result["error_message"],
                     },
                 )
-
-            # Test if date columns have date format
-            if core.utils.load_batch.check_format_date(sample_batch_df):
+            # format date columns
+            sample_batch_df = core.utils.load_batch.check_and_format_date(
+                sample_batch_df
+            )
+            if not isinstance(sample_batch_df, pd.DataFrame):
                 return render(
                     request,
                     "wetlab/record_sample.html",
                     {
                         "fields_info": fields_info,
-                        "error_message": core.utils.load_batch.check_format_date(
-                            sample_batch_df
-                        ),
+                        "error_message": sample_batch_df,
                     },
                 )
-
-            # Reformat date columns to correct format (mandatory before converting to json)
-            sample_batch_df = core.utils.load_batch.format_date(sample_batch_df)
-
+            sample_batch_df = core.utils.load_batch.heading_refactor(sample_batch_df)
```

Comment:
If you use this before, as in the previous code, you don't need to change to field_name to verbose name in the validate header function

Reply:
Sorry , but I this time I am completed lost about what you mean